composer.json

```diff
@@ -16,18 +16,17 @@
     "dealerdirect/phpcodesniffer-composer-installer": "0.7.0",
     "php-parallel-lint/php-parallel-lint": "^1.2",
     "phpcompatibility/phpcompatibility-wp": "2.1.0",
+    "phpunit/phpunit": "^5 || ^6 || ^7 || ^8 || ^9",
     "roave/security-advisories": "dev-master",
     "sirbrillig/phpcs-variable-analysis": "2.9.0",
-    "squizlabs/php_codesniffer": "^3"
+    "squizlabs/php_codesniffer": "^3",
+    "yoast/phpunit-polyfills": "^0.2.0"
   },
   "suggest": {
     "ext-json": "Provides native implementation of json_encode()/json_decode().",
     "ext-mbstring": "Used by Dom\\Document to convert encoding to UTF-8 if needed."
   },
   "config": {
-    "platform": {
-      "php": "5.6"
-    },
```

Comment:
I suppose removing this allows for PHPUnit versions > 6.0 to be installed. Is there any other way we could enforce PHP 5.6 code compatibility.? I'm not sure if that's possible wit PHPStan.

Reply:
The platform configuration is irrelevant here, as it is a library, not an end project. We don't commit the lock file, and the platform setting will be ignored when the library is not the root project.

Libraries are only meant to provide constraints from within which they are supposed to work. And we already do so above in the `require` section.

This `platform` setting would be important for the plugin, though, where we actually rely on the result of the Composer resolution, instead of only defining the constraints. As the plugin in its _built_ form is meant to be compatible from PHP 5.6 all the way up to PHP 8, we have to only use code that runs on the lowest possible common constraints. That's where the `platform` setting is then used, so that you don't build against the PHP version you're running Composer with, but you built against the lowest version of PHP that is still meant to be able to run the built artifact.

Reply:
Note that, as we have removed the platform setting now, the library will be built with different versions of its dependencies on different versions of the PHP testing matrix. And that is to be expected.

We should probably also add a `composer update --prefer-lowest` version to the test matrix, as Composer by default will prefer the highest available version.

Reply:
> The platform configuration is irrelevant here, as it is a library, not an end project. We don't commit the lock file, and the platform setting will be ignored when the library is not the root project.

Ah I see, thanks for the explanation!